src/frontend/planner_test/tests/testdata/shared_views.yaml

```diff
@@ -0,0 +1,21 @@
+- sql: |
+    create table t1 (x int, y int);
+    create view v1 as select x + y as z from t1 where y > 0;
+    create view v2 as select x * z as a, y * z as b from t1, v1 where z = x;
+    select * from v1, v2;
+  logical_plan: |
+    LogicalProject { exprs: [(t1.x + t1.y), (t1.x * (t1.x + t1.y)), (t1.y * (t1.x + t1.y))] }
+    └─LogicalJoin { type: Inner, on: true, output: all }
+      ├─LogicalShare { id = 4 }
+      | └─LogicalProject { exprs: [(t1.x + t1.y)] }
+      |   └─LogicalFilter { predicate: (t1.y > 0:Int32) }
+      |     └─LogicalScan { table: t1, columns: [t1.x, t1.y, t1._row_id] }
+      └─LogicalShare { id = 9 }
+        └─LogicalProject { exprs: [(t1.x * (t1.x + t1.y)), (t1.y * (t1.x + t1.y))] }
+          └─LogicalFilter { predicate: ((t1.x + t1.y) = t1.x) }
+            └─LogicalJoin { type: Inner, on: true, output: all }
+              ├─LogicalScan { table: t1, columns: [t1.x, t1.y, t1._row_id] }
+              └─LogicalShare { id = 4 }
+                └─LogicalProject { exprs: [(t1.x + t1.y)] }
+                  └─LogicalFilter { predicate: (t1.y > 0:Int32) }
+                    └─LogicalScan { table: t1, columns: [t1.x, t1.y, t1._row_id] }
```

Comment:
How about adding a `stream_plan` here to ensure this logical plan can be converted to an expected `stream_plan`? BTW, we also need a e2e test.

Reply:
Sure! I'll update the test and rebase with main later.

Reply:
I've updated with a new `stream_plan`, but I also need to include a filter on the join to avoid the error on not supporting nested loop join in streaming mode.
An e2e test is also added.